### PR TITLE
Add Apache Camel VS Code Extension (#109)

### PR DESCRIPTION
**Pull request by apupier**
Signed-off-by: Aurélien Pupier <apupier@redhat.com>

create Pull request to check if tests are passing.

How and where to create automated test checking that the plugin is working correctly in Che?

### What does this PR do?

this PR is adding Language Support for Camel to the list of extension

### CODE REVIEW

**Comment by apupier** [comment]
failure due to only svg are supported as icon:
```
!!!   Wrong icon type found in 'camel-tooling.vscode-apache-camel/0.0.14':
!!!   'https://github.com/camel-tooling/camel-lsp-client-vscode/raw/0.0.14/icons/icon128.png' Make sure it is pointing to .svg image.
```

File: plugins/camel-tooling.vscode-apache-camel/0.0.14/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+title: Language Support for Apache Camel
+description: This VS Code extension provides support for Apache Camel.
+url: https://marketplace.visualstudio.com/_apis/public/gallery/publishers/camel-tooling/vsextensions/vscode-apache-camel/0.0.14/vspackage
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
+publisher: Camel Tooling
+repository: https://github.com/camel-tooling/camel-lsp-client-vscode
+category: Language
+firstPublicationDate: "2018-04-23"
+attributes:
+  containerImage: "eclipse/che-remote-plugin-runner-java8:next"
```
Comment (ibuziuk) [inline plugins/camel-tooling.vscode-apache-camel/0.0.14/meta.yaml]: no new line in the end

Comment (apupier) [reply plugins/camel-tooling.vscode-apache-camel/0.0.14/meta.yaml]: new line added

**Comment by apupier** [comment]
I guess this PR will require to use a Github hosted vsix too.
Anything else which has prevented it to be merged?

**Comment by ibuziuk** [comment]
the rest looks good

**Comment by apupier** [comment]
using the direct link vsix to github is not working.
The installation seems to go fine. There is even the extension listed in plugins view but it is like the extension is not activated. There is no completion or hover. There is no output channel for the Language server created.

**Comment by ibuziuk** [comment]
@apupier are you sure the binaries are the same? I'm not aware of anything `url` specific on plugin broker processing side

**Comment by apupier** [comment]
I used the vsix from the VSCode marketplace.
I double-checked that it is the correct vsix that i uploaded.
I redownloaded the vsix from github and installed it in a local VS Code instance and it is working fine

**Comment by apupier** [comment]
this line is quite strange:

`root INFO Deploying backend plugin "vscode-apache-camel@0.0.14" from "/plugins/camel-tooling.vscode-apache-camel.0.0.14.cameltooling_vscodeapachecamel/extension/out/extension"`

I would expect to have the extension.js like for other plugins.

I checked that in filesystem, there is only a file named extension.js at this place

**Comment by benoitf** [comment]
> this line is quite strange:
> root INFO Deploying backend plugin "vscode-apache-camel@0.0.14" from "/plugins/camel-tooling.vscode-apache-camel.0.0.14.cameltooling_vscodeapachecamel/extension/out/extension"
>
>I would expect to have the extension.js like for other plugins.

vsix extensions package files in a subfolder extension so when unpacking files there are extra files but at the end it load the correct entry-point

**Comment by apupier** [comment]
I launched a third time and now the extension is working even with the github vsix url.
I notice no differences in the log compared to when it wasn't.

**Comment by garagatyi** [comment]
Please note that I added CircleCI and Eclipse sign off check to required checks for PRs.
Since this PR was created before CircleCI was configured in the repository you have to commit something to this PR to initiate CircleCI build that is required to proceed with merging the PR.
Sorry for the inconvenience.

**Comment by apupier** [comment]
I noticed that several other more recent PR have been merged.
Is something preventing merge of this PR?

**Comment by ibuziuk** [comment]
@apupier looks like you now also need to add the plugin to the https://github.com/eclipse/che-plugin-registry/tree/master/v2/plugins to have it available for selection once all the changes related to #55 are applied. But this should not block merging the PR @garagatyi wdyt?

**Comment by apupier** [comment]
I will prepare a commit with the modification  for the v2.
Let me know if I add it to this branch or if it is better to create another PR.

**Comment by apupier** [comment]
I prepared another branch with v2 registry too https://github.com/apupier/che-plugin-registry/tree/109-addforregistryv1andV2 let me know how you prefer to proceed.

**Comment by garagatyi** [comment]
@apupier @ibuziuk I'm OK with either variant

**Comment by apupier** [comment]
added on top of this PR then.

Shellcheck is failing due to not related change to this PR. Mentioned to ignore it in this other PR comment https://github.com/eclipse/che-plugin-registry/pull/124#issuecomment-486685160

**Comment by ibuziuk** [comment]
@apupier merged but you will need to add it in v3 also now due to - Merge meta.yaml and plugin yaml change - https://github.com/eclipse/che-plugin-registry/pull/135

**Comment by bfitzpat** [comment]
> @apupier merged but you will need to add it in v3 also now due to - Merge meta.yaml and plugin yaml change - #135

Will take a look

**Pull request merged**